Core/MIPS/IR/IRInst.cpp

```diff
@@ -0,0 +1,589 @@
+					mips->fs[inst->dest] = 2147483647LL;
+				}
+			} else {
+				// Overflow happens to be the right value anyway.
+				mips->fs[inst->dest] = (int)ceilf(src);
+			}
+			break;
+		}
+		case IROp::FCeil:
+			mips->fs[inst->dest] = (int)ceilf(mips->f[inst->src1]);
+			break;
+		case IROp::FFloor:
+			mips->fs[inst->dest] = (int)floorf(mips->f[inst->src1]);
+			break;
+
+		case IROp::FCvtSW:
+			mips->f[inst->dest] = (float)mips->fs[inst->src1];
+			break;
+		case IROp::FCvtWS:
+		{
+			float src = mips->f[inst->src1];
+			if (my_isnanorinf(src))
+			{
+				mips->fs[inst->dest] = my_isinf(src) && src < 0.0f ? -2147483648LL : 2147483647LL;
+				break;
+			}
+			switch (mips->fcr31 & 3)
+			{
+			case 0: mips->fs[inst->dest] = (int)round_ieee_754(src); break;  // RINT_0
+			case 1: mips->fs[inst->dest] = (int)src; break;  // CAST_1
+			case 2: mips->fs[inst->dest] = (int)ceilf(src); break;  // CEIL_2
+			case 3: mips->fs[inst->dest] = (int)floorf(src); break;  // FLOOR_3
+			}
+			break; //cvt.w.s
+		}
+		case IROp::ZeroFpCond:
+			mips->fpcond = 0;
+			break;
+
+		case IROp::FMovFromGPR:
+			memcpy(&mips->f[inst->dest], &mips->r[inst->src1], 4);
+			break;
+		case IROp::FMovToGPR:
+			memcpy(&mips->r[inst->dest], &mips->f[inst->src1], 4);
+			break;
+
+		case IROp::ExitToConst:
+			return constPool[inst->dest];
+
+		case IROp::ExitToReg:
+			return mips->r[inst->dest];
+
+		case IROp::ExitToConstIfEq:
+			if (mips->r[inst->src1] == mips->r[inst->src2])
+				return constPool[inst->dest];
+			break;
+		case IROp::ExitToConstIfNeq:
+			if (mips->r[inst->src1] != mips->r[inst->src2])
+				return constPool[inst->dest];
+			break;
+		case IROp::ExitToConstIfGtZ:
+			if ((s32)mips->r[inst->src1] > 0)
+				return constPool[inst->dest];
+			break;
+		case IROp::ExitToConstIfGeZ:
+			if ((s32)mips->r[inst->src1] >= 0)
+				return constPool[inst->dest];
+			break;
+		case IROp::ExitToConstIfLtZ:
+			if ((s32)mips->r[inst->src1] < 0)
+				return constPool[inst->dest];
+			break;
+		case IROp::ExitToConstIfLeZ:
+			if ((s32)mips->r[inst->src1] <= 0)
+				return constPool[inst->dest];
+			break;
+
+		case IROp::Downcount:
+			mips->downcount -= (inst->src1) | ((inst->src2) << 8);
+			break;
+
+		case IROp::SetPC:
+			mips->pc = mips->r[inst->src1];
+			break;
+
+		case IROp::SetPCConst:
+			mips->pc = constPool[inst->src1];
+			break;
+
+		case IROp::Syscall:
+			// SetPC was executed before.
+		{
+			MIPSOpcode op(constPool[inst->src1]);
+			CallSyscall(op);
+			return mips->pc;
+		}
+
+		case IROp::Interpret:  // SLOW fallback. Can be made faster.
+		{
+			MIPSOpcode op(constPool[inst->src1]);
+			MIPSInterpret(op);
+			break;
+		}
+
+		case IROp::CallReplacement:
+		{
+			int funcIndex = constPool[inst->src1];
+			const ReplacementTableEntry *f = GetReplacementFunc(funcIndex);
+			int cycles = f->replaceFunc();
+			mips->downcount -= cycles;
+			return mips->r[MIPS_REG_RA];
+		}
+
+		case IROp::Break:
+			Crash();
```

Comment:
Definitely want this one to be optional, at least when fastmem is off.

-[Unknown]